pkg/iostreams/regex_filter_writer.go

```diff
@@ -0,0 +1,46 @@
+package iostreams
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"regexp"
+)
+
+func NewRegexFilterWriter(out io.Writer, regexp *regexp.Regexp, repl string) io.Writer {
+	return &RegexFilterWriter{out: out, regexp: *regexp, repl: repl}
+}
+
+type RegexFilterWriter struct {
```

Comment:
This name seems verbose, so if there are any other name ideas I am open to them.